src/client/app/containers/BarChartContainer.js

```diff
@@ -35,6 +35,24 @@ function mapStateToProps(state) {
 			}
 		}
 	}
+
+	for (const groupID of state.graph.selectedGroups) {
```

Comment:
Could you refactor this loop and the loop above it (for meters) into a function? You duplicate a lot of code.

Reply:
I can, but I'd like to do a larger refactor to use datasource ID objects later and it seems like wasted effort to refactor twice.